jep/README.md

```diff
@@ -0,0 +1,142 @@
+This is one of the main pain point for external parties that are interested in integrating some JupyterLab features in their products.
+- [Web Components](https://developer.mozilla.org/en-US/docs/Web/Web_Components) "is a suite of different technologies allowing you to create reusable custom elements — with their functionality encapsulated away from the rest of your code — and utilize them in your web apps" (quote from Mozilla developer).  
+They are built on the [custom element](https://html.spec.whatwg.org/multipage/custom-elements.html#dom-window-customelements) web standard.
+- A `Reactive UI` is not available today in Jupyter and can be defined as ... (see https://en.wikipedia.org/wiki/Reactive_user_interface, better defintion?). The best known example of such a UI is [ObservableHQ](https://observablehq.com).
+- A `global state` is not provided by the "advanced framework". React.js developers usually have to choose from existing solutions like [Redux](https://redux.js.org), [Mobx](https://mobx.js.org). For Vue.js, [Vuex](https://vuex.vuejs.org) is an option.
+- `Local-first`: Read more about the Local-first applications [in this article](https://www.inkandswitch.com/local-first), "a set of principles for software that enables both collaboration and ownership for users".
+
+The current usage of those technologies at Jupyter is:
+
+- Notebook until V6: Relies on jQuery.
+- Notebook V7: Relies on Retrolab, thus on Lumino and JupyterLab (https://jupyter.org/enhancement-proposals/79-notebook-v7/notebook-v7.html).
+- JupyterLab: Relies on Lumino, potentially with React.js underneath.
+- JupyterHub: Relies on jQuery and React.js.
+- Nteract: Relies on React.js.
+
+The goal is to create accessible themed components that is usable by developers (with a focus on _React.js_ usage), while being **Reactive**,  **Collaborative** and easy to integrate outside official Jupyter projects. The migration plan from an existing implementation has to be **non-breaking** and **progressive** (non need to switch everything at once).
+
+# Reference-level explanation
+
+We propose to build the widget toolkit on the current work being done at https://github.com/jupyterlab-contrib/jupyter-ui-toolkit. The features of that library include:
+
+- **Implements the Jupyter design language:** All components follow the design language of Jupyter
+  – enabling developers to create extensions that have a consistent look and feel with the rest of
+  the ecosystem.
+- **Automatic support for color themes:** All components are designed with theming in mind and will
+  automatically display the current application theme.
+- **Use any tech stack:** The library ships as a set of web components, meaning developers can use
+  the toolkit no matter what tech stack (React, Vue, Svelte, etc.) their extension is built with.
+- **Accessible out of the box:** All components ship with web standard compliant ARIA labels,
```

Comment:
I don't know that we can guarantee everything will be accessible without any edits since these components could be used in a number of contexts. I still want to highlight the wins, just being more specific.

```suggestion
- **Accessible foundations out of the box:** All components ship with web standard compliant ARIA labels,
```